src/settypes/recursiveset.jl

```diff
@@ -3,7 +3,7 @@
 
 Lazy union of sets.
 """
-struct RecursiveSet{T} <: AbstractSet{T}
+mutable struct RecursiveSet{T} <: AbstractSet{T}
```

Comment:
Why was this made mutable? To support `union!`?

Reply:
Yes, because otherwise in-place `union!` is not possible. But to be fair I don't think this one is useful for Hessians anyway, because there is no fast way to construct a Cartesian product